### PR TITLE
testhelper.py: Improve _update_gpu_tag for same failure on different GPUs

### PR DESCRIPTION
**Pull request by haoxli**
Improve the function to avoid conflicts when a test case have two expectations for different Intel devices on same OS platform.
- Skip updating the expectations of other vendors and mac platforms, we just update win/linux specific and Intel specific failures.
- Add OS, adapter, dxc, graphite to check for duplication of the updated  expectations.

### CODE REVIEW

**Comment by haoxli** [comment]
webgpu adapter and dxc tags also can make conflicts.

File: misc/testhelper.py [hunk 1]
```diff
@@ -109,6 +109,10 @@ class TestExpectation:
             'crbug.com/0000 [ intel win ] webgpu:shader,execution,memory_layout:write_layout:case="mat4x4f_size";aspace="workgroup" [ Failure ]',
             'crbug.com/0000 [ intel win ] webgpu:shader,execution,memory_layout:write_layout:case="mat4x4h_align8";aspace="workgroup" [ Failure ]',
             'crbug.com/0000 [ intel win ] webgpu:shader,execution,memory_layout:write_layout:case="mat4x4h_size";aspace="workgroup" [ Failure ]',
+            # https://issues.chromium.org/issues/42240657
+            # This expectation exists in webgpu-cts/expectations.txt, but only for Intel Gen9 GPUs which graphite is disabled.
+            # On Intel Gen12 GPUs, graphite is enabled, we need another expectation with graphite-enabled for the failure.
+            'crbug.com/dawn/1706 [ graphite-enabled intel win ] webgpu:api,operation,command_buffer,queries,occlusionQuery:occlusion_query,depth:writeMask=0;* [ Failure ]',
```
Comment (gyagp) [inline misc/testhelper.py:115]: Why do we hit this issue when using graphite, while upstream doesn't have it?

File: misc/testhelper.py [hunk 2]
```diff
@@ -109,6 +109,10 @@ class TestExpectation:
             'crbug.com/0000 [ intel win ] webgpu:shader,execution,memory_layout:write_layout:case="mat4x4f_size";aspace="workgroup" [ Failure ]',
             'crbug.com/0000 [ intel win ] webgpu:shader,execution,memory_layout:write_layout:case="mat4x4h_align8";aspace="workgroup" [ Failure ]',
             'crbug.com/0000 [ intel win ] webgpu:shader,execution,memory_layout:write_layout:case="mat4x4h_size";aspace="workgroup" [ Failure ]',
+            # https://issues.chromium.org/issues/42240657
+            # This expectation exists in webgpu-cts/expectations.txt, but only for Intel Gen9 GPUs which graphite is disabled.
```
Comment (gyagp) [inline misc/testhelper.py:113]:
```suggestion
            # This expectation exists in webgpu-cts/expectations.txt, but only for Intel Gen9 GPUs where graphite is disabled.
```

File: misc/testhelper.py [hunk 3]
```diff
@@ -178,32 +211,61 @@ def _update_gpu_tag(line, intel_expectations=[]):
...
+        # No update is needed, return the original content.
+        if new_tags == tags:
+            return line
 
-        # If the updated line already exists, just comment the line,
+        # If the updated test case already exists, just comment the line,
         # otherwise comment the line and append the updated line.
-        # For the line already with 'intel' tag, keep as is if there is no duplicate line.
-        if new_line in intel_expectations:
+        check_test = f'{test_os},{webgpu_adapter},{webgpu_dxc},{graphite},{test_name}'
```
Comment (gyagp) [inline misc/testhelper.py:262]: Why do we need to track all kinds of tags? Will they have different order?

File: misc/testhelper.py (hunk 1 above)
Comment (haoxli) [reply misc/testhelper.py:115]: This expectation was 
> crbug.com/dawn/1706 [ intel ] 
webgpu:api,operation,command_buffer,queries,occlusionQuery:occlusion_query,depth:writeMask=0;* [ Failure ]

recently upstream updated it to 
> crbug.com/dawn/1706 [ angle-d3d11 **graphite-disabled** intel-0x9bc5 release renderer-skia-gl win10 ] webgpu:api,operation,command_buffer,queries,occlusionQuery:occlusion_query,depth:writeMask=0;* [ Failure ]

On Gen9 bots, graphite is unavaiable and disabled, this failure can run as expected. But on Gen12, graphite is enabled, this expectation does not work.  we need add it here.

While upstream bots enable Gen12 machines, they will also need this expectation. after that, we can remove it.

File: misc/testhelper.py (hunk 3 above)
Comment (haoxli) [reply misc/testhelper.py:262]: No order required. they are used to check whether a failure is duplicated. 

`[ linux webgpu-dxc-enabled webgpu-adapter-default graphite-disabled ] test1`
`[ linux webgpu-dxc-disabled webgpu-adapter-default graphite-disabled ] test1`
`[ linux webgpu-dxc-enabled webgpu-adapter-swiftshader graphite-disabled ] test1`
`[ linux webgpu-dxc-enabled webgpu-adapter-default graphite-enabled ] test1`

They are available expectations for different failures.

**Comment by haoxli** [comment]
Resolve the conflicts in https://github.com/webatintel/toolkit/pull/57.

**Pull request closed without merging**